pkg/controller/node/nodecontroller.go

```diff
@@ -740,35 +740,37 @@ func (nc *NodeController) tryUpdateNodeStatus(node *v1.Node) (time.Duration, v1.
 			if observedReadyCondition.Status != v1.ConditionUnknown {
 				currentReadyCondition.Status = v1.ConditionUnknown
 				currentReadyCondition.Reason = "NodeStatusUnknown"
-				currentReadyCondition.Message = fmt.Sprintf("Kubelet stopped posting node status.")
+				currentReadyCondition.Message = "Kubelet stopped posting node status."
 				// LastProbeTime is the last time we heard from kubelet.
 				currentReadyCondition.LastHeartbeatTime = observedReadyCondition.LastHeartbeatTime
 				currentReadyCondition.LastTransitionTime = nc.now()
 			}
 		}
 
-		// Like NodeReady condition, NodeOutOfDisk was last set longer ago than gracePeriod, so update
-		// it to Unknown (regardless of its current value) in the master.
-		// TODO(madhusudancs): Refactor this with readyCondition to remove duplicated code.
-		_, oodCondition := v1.GetNodeCondition(&node.Status, v1.NodeOutOfDisk)
-		if oodCondition == nil {
-			glog.V(2).Infof("Out of disk condition of node %v is never updated by kubelet", node.Name)
-			node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{
-				Type:               v1.NodeOutOfDisk,
-				Status:             v1.ConditionUnknown,
-				Reason:             "NodeStatusNeverUpdated",
-				Message:            fmt.Sprintf("Kubelet never posted node status."),
-				LastHeartbeatTime:  node.CreationTimestamp,
-				LastTransitionTime: nc.now(),
-			})
-		} else {
-			glog.V(4).Infof("node %v hasn't been updated for %+v. Last out of disk condition is: %+v",
-				node.Name, nc.now().Time.Sub(savedNodeStatus.probeTimestamp.Time), oodCondition)
-			if oodCondition.Status != v1.ConditionUnknown {
-				oodCondition.Status = v1.ConditionUnknown
-				oodCondition.Reason = "NodeStatusUnknown"
-				oodCondition.Message = fmt.Sprintf("Kubelet stopped posting node status.")
-				oodCondition.LastTransitionTime = nc.now()
+		// remaining node conditions should also be set to Unknown
+		remainingNodeConditionTypes := []v1.NodeConditionType{v1.NodeOutOfDisk, v1.NodeMemoryPressure, v1.NodeDiskPressure, v1.NodeNetworkUnavailable}
```

Comment:
I'm not sure about NetworkUnavailable - IIRC this is basically 'route is created' condition, so Kubelet isn't involved in setting this. @wojtek-t 

Rest seems fine.